python/TestHarness/tests/test_resultssummary.py

```diff
@@ -0,0 +1,148 @@
+#* This file is part of the MOOSE framework
+#* https://mooseframework.inl.gov
+#*
+#* All rights reserved, see COPYRIGHT for full restrictions
+#* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+#*
+#* Licensed under LGPL 2.1, please see LICENSE for details
+#* https://www.gnu.org/licenses/lgpl-2.1.html
+
+import unittest
+import os
+from mock import patch
+from contextlib import redirect_stdout
+from io import StringIO
+
+from TestHarness.resultsreader.reader import TestHarnessResultsReader
+from TestHarness.resultssummary.summary import TestHarnessResultsSummary
+from TestHarness.resultsreader.results import TestName
+
+# Whether or not authentication is available from env var RESULTS_READER_AUTH_FILE
+HAS_AUTH = TestHarnessResultsReader.hasEnvironmentAuthentication()
+
+# Test database name for testing pull request results
+TEST_DATABASE_NAME = 'civet_tests_moose_test_results'
+# Arguments for using the production database for getting real results
+TEST_FOLDER_NAME = 'tests/test_harness'
+TEST_TEST_NAME = 'ok'
+TEST_NAME = TestName(TEST_FOLDER_NAME, TEST_TEST_NAME)
+
+EVENT_ID = os.environ.get('CIVET_EVENT_ID')
+if EVENT_ID is not None:
+    EVENT_ID = int(EVENT_ID)
+
+class TestResultsSummary(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @unittest.skipUnless(HAS_AUTH, "Skipping because authentication is not available")
```

Comment:
Similarly, we should have a simple doc string for each test method that describes what the test is testing